pkg/webhook/nodenetworkconfigurationpolicy/handler.go

```diff
@@ -43,6 +48,56 @@ func mutatePolicyHandler(neededMutationFor func(nmstatev1beta1.NodeNetworkConfig
 	}
 }
 
+func validatePolicyHandler(cli client.Client, neededValidationFor func(nmstatev1beta1.NodeNetworkConfigurationPolicy, nmstatev1beta1.NodeNetworkConfigurationPolicy) bool, validators ...validator) admission.HandlerFunc {
+	log := logf.Log.WithName("webhook/nodenetworkconfigurationpolicy/validator")
+	return func(ctx context.Context, req webhook.AdmissionRequest) webhook.AdmissionResponse {
+		original := req.Object.Raw
+		newPolicy := nmstatev1beta1.NodeNetworkConfigurationPolicy{}
+		err := json.Unmarshal(original, &newPolicy)
+		if err != nil {
+			return admission.Errored(http.StatusInternalServerError, errors.Wrapf(err, "failed decoding policy: %s", string(original)))
+		}
+		currentPolicy := nmstatev1beta1.NodeNetworkConfigurationPolicy{}
+		err = cli.Get(context.TODO(), types.NamespacedName{Name: newPolicy.GetName(), Namespace: newPolicy.GetNamespace()}, &currentPolicy)
+		if err != nil && !apierrors.IsNotFound(err) {
+			errMsg := fmt.Sprintf("failed getting policy %s", string(original))
+			log.Error(err, errMsg)
+			return admission.Errored(http.StatusInternalServerError, errors.Wrapf(err, errMsg))
+		}
+
+		if !neededValidationFor(newPolicy, currentPolicy) {
+			return admission.Allowed("validation not needed")
+		}
+
+		policyList := nmstatev1beta1.NodeNetworkConfigurationPolicyList{}
+		err = cli.List(context.TODO(), &policyList)
+		if err != nil {
+			errMsg := "unable to validate, failed listing NodeNetworkConfigurationPolicies"
+			log.Error(err, errMsg)
+			return admission.Errored(http.StatusInternalServerError, errors.Wrap(err, errMsg))
+		}
+
+		errCauses := []metav1.StatusCause{}
+		for _, policy := range policyList.Items {
+			for _, validate := range validators {
+				errCauses = append(errCauses, validate(policy)...)
+			}
+		}
+		if len(errCauses) > 0 {
+			return admission.Denied(handlePolicyCauses(errCauses, newPolicy.Name))
+		}
+		return admission.Allowed("")
+	}
+}
+
+func handlePolicyCauses(causes []metav1.StatusCause, name string) string {
+	errMsg := fmt.Sprintf("Failed to admit NodeNetworkConfigurationPolicy %s: ", name)
```

Comment:
following the same error messages, maybe keep the first word uncapitalized: `Failed to`-->`failed to`

Reply:
Done